spring-integration-core/src/main/java/org/springframework/integration/dsl/RouterSpec.java

```diff
@@ -42,8 +42,7 @@
  * @since 5.0
  */
 public final class RouterSpec<K, R extends AbstractMappingMessageRouter>
-		extends AbstractRouterSpec<RouterSpec<K, R>, R>
-		implements ComponentsRegistration {
+		extends AbstractRouterSpec<RouterSpec<K, R>, R> {
```

Comment:
How do you catch this kind of issues?

Thanks

Reply:
I enabled an eclipse compiler error/warning option:

```
Description	Resource	Path	Location	Type
Redundant superinterface ComponentsRegistration for the type RouterSpec<K,R>, already defined by EndpointSpec<RouterSpec<K,R>,ConsumerEndpointFactoryBean,R>	RouterSpec.java	/spring-integration-core/src/main/java/org/springframework/integration/dsl	line 46	Java Problem
```